### ISSUE TITLE
Qemu only : linux/arm/v7 'Command 'install' is not valid, try 'pecl help'

### ISSUE THREAD

**Issue by ohnotnow** [moby/qemu#8]
Resubmitting from https://github.com/docker-library/php/issues/893 .

Slightly odd issue - if I try and build a multiarch image and use a `pecl install redis` as part of it - then it fails on the linux/arm/v7 build.  Example Dockerfile :
```
FROM php:7.3-apache
RUN pecl install redis && docker-php-ext-enable redis
```

Running : 
```
docker buildx build --pull --load --no-cache --platform linux/arm/v7 -t test-pecl -f Dockerfile .
```
Fails with 
```
 => ERROR [2/2] RUN pecl install redis && docker-php-ext-enable redis      1.3s
------                                                                          
 > [2/2] RUN pecl install redis && docker-php-ext-enable redis:                 
#5 0.583 Command 'install' is not valid, try 'pecl help' 
```

Running :
```
docker buildx build --pull --load --no-cache --platform linux/amd64 -t test-pecl -f Dockerfile .
```
Works fine. The docker version in front of me is `Docker version 19.03.2, build 6a30dfc`.  Let me know if there's any more info I can supply or something to dig into :-)

Oh - and running the php:7.3 image on an actual ARM machine also runs `pecl install redis` a-ok as noted on the original issue linked above.

**Comment by ohnotnow** [comment]
Oddly - if I run the same image by hand, the `pecl install redis` runs fine.  Eg,
```
docker run -it --rm --platform linux/arm/v7 php:7.3-apache bash
root@4b542d27ebea:/var/www/html# pecl install redis
# success
```
So it _could_ be a buildx issue I guess.  Or not.  :: waves hands vaguely ::

**Comment by StefanScherer** [comment]
Triaging this problem, I could reproduce the problem in
- Docker Desktop 2.1.0.2 (Docker engine 19.03.2) on Mac (kernel 4.9.184-linuxkit)
- a boot2docker 18.09.9 docker-machine in VirtualBox on the same Mac (kernel 4.19)
- a VMware Fusion Vagrant ubuntu box (kernel 4.15.0)

Interesting, I could run the build on CircleCI (docker 17.09.0, kernel 4.15.0) https://circleci.com/gh/StefanScherer/circle2test/81?pipelines-ui-opt-out in a quick-and-dirty test environment.

In all non-DockerDesktop environments I used this image to register QEMU

```
docker run --rm --privileged hypriot/qemu-register
```

and this slightly adjusted Dockerfile to explicitly pull the ARM image for a "normal" `docker build -t phptest .`.

```Dockerfile
FROM arm32v7/php:7.3-apache
RUN pecl install redis && docker-php-ext-enable redis
```

TL;DR It doesn't work on my MacBook, but it works on CircleCI. Different host CPU? 🤔

**Comment by JnMik** [comment]
Same issue here running 

`yes | pecl install apcu`

in this docker file
```
FROM biarms/qemu-bin as qemu-bin-ref
FROM arm32v7/php:7.3-fpm
COPY --from=qemu-bin-ref /usr/bin/qemu-arm-static /usr/bin/qemu-arm-static
RUN yes | pecl install apcu

```

Will fail in the build phase with the error mentioned above

Edit: Mine fails on gitlab pipeline running in docker:19.03.1-dind I believe

**Comment by JnMik** [comment]
Anyone found a workaround ?
